src/inmanta/data/__init__.py

```diff
@@ -1972,6 +2001,86 @@ async def get_next_version(self) -> int:
         self.last_version = version
         return version
 
+    @classmethod
+    async def get_list(
+        cls,
+        order_by_column: Optional[str] = None,
+        order: str = "ASC",
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+        no_obj: bool = False,
+        connection: Optional[asyncpg.connection.Connection] = None,
+        details: bool = True,
+        **query: object,
+    ) -> List["BaseDocument"]:
+        """
+        Get a list of documents matching the filter args.
+
+        """
+        if details:
+            return await super().get_list(
+                order_by_column=order_by_column,
+                order=order,
+                limit=limit,
+                offset=offset,
+                no_obj=no_obj,
+                connection=connection,
+                **query,
+            )
+        return await cls.get_list_without_details(
+            order_by_column=order_by_column,
+            order=order,
+            limit=limit,
+            offset=offset,
+            no_obj=no_obj,
+            connection=connection,
+            **query,
+        )
+
+    @classmethod
+    async def get_list_without_details(
+        cls,
+        order_by_column: Optional[str] = None,
+        order: str = "ASC",
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+        no_obj: bool = False,
+        connection: Optional[asyncpg.connection.Connection] = None,
+        **query: object,
+    ) -> List["BaseDocument"]:
+        """
+        Get a list of environments matching the filter args.
+        Don't return the description and icon columns.
+        """
+        columns = ["id", "name", "project", "repo_url", "repo_branch", "settings", "last_version", "halted"]
```

Comment:
The problem with this implementation is that we will have to add any new columns of the `Environment` table to this list as well. It's likely that we will forget to do that. Would it be possible to implement this the other way around? I mean, dynamically retrieve all column names in the `Environment` table and remove the `icon` and the `description` from them.

Reply:
Yes, good idea